aspnetcore/client-side/blazor/introduction/index.md

```diff
@@ -0,0 +1,130 @@
+
+```cshtml
+<div>
+    <h2>@Title</h2>
+    @RenderContent(Body)
+    <button onclick=@OnOK>OK</button>
+</div>
+
+@functions {
+    public string Title { get; set; }
+    public Content Body { get; set; }
+    public Action OnOK { get; set; }
+}
+```
+
+When this component is used elsewhere in the app, IntelliSense speeds development with syntax completion and parameter info.
+
+Components can be:
+
+* Nested.
+* Generated procedurally in code.
+* Shared via class libraries.
+* Unit tested without requiring a browser DOM.
+
+## Infrastructure
+
+Blazor offers the core facilities that most apps require, including:
+
+* Layouts
+* Routing
+* Dependency injection
+* Lazy loading (loading parts of the app on demand as a user navigates the app)
+* Unit testing
+
+All of these features are optional. When one of these features isn't used in an app, the implementation is stripped out of the app when published by the IL linker.
+
+A few low-level elements are included in the framework. For example, routing and layouts aren't built-in. Routing and layouts are implemented in *user space*, code that an app developer can write without using internal APIs. These features can be replaced with different systems to suit the app's requirements. The current layouts prototype is implemented in about 30 lines of C# code, so a developer can understand and replace it if desired.
+
+## Code sharing and .NET Standard
+
+The [.NET Standard](/dotnet/standard/net-standard) is a formal specification of .NET APIs that are intended to be available on all .NET implementations. Mono on WebAssembly supports `netstandard2.0` or higher. .NET Standard class libraries can be shared across server code and in browser-based apps.
+
+Browsers support the APIs that developers use to build web apps. Not all .NET APIs are callable from the browser. For example, arbitrary TCP sockets can't be accessed in a browser, so [System.Net.Sockets.TcpListener](/dotnet/api/system.net.sockets.tcplistener) can't perform any useful task. For BCL APIs that don't apply to a given platform, the BCL throws a [PlatformNotSupported](/dotnet/api/system.platformnotsupportedexception) exception.
+
+## JavaScript/TypeScript interop
+
+For apps that require third-party JavaScript libraries and browser APIs, WebAssembly is designed to interoperate with JavaScript. Blazor is capable of using any library or API that JavaScript is able to use. The Mono team is working on a library that exposes standard browser APIs to .NET.
+
+## Optimization
+
+Traditionally, .NET has focused on platforms where the app's binary size isn't a major concern. It doesn't really matter whether a server-side ASP.NET app is 1MB or 50MB. It's only a moderate concern for native desktop or mobile apps. But for browser apps, payload size is critical.
```

Comment:
I recommend changing "browser apps" to something like "client-side apps". Sure, ASP.NET runs on the server, but it also is served up in the browser.

Reply:
Maybe I'm misreading your reasoning, but I like your suggestion better because browsers run on servers, too! :smile: "Client-side" makes it clear we're talking about sending the app somewhere else.